### PR TITLE
Backend timeout

### LINKED ISSUE #368: Implement timeouts for mock backend subscription
Side node: As the MockBackend used for testing the core does not use proper time out logic yet, these things are easily missed in the core client tests. We should implement the timeout logic for events emitted by the Mock Backend Subscription.

_Originally posted by @matthiasgeihs in https://github.com/hyperledger-labs/go-perun/issues/367#issuecomment-1203132061_

### PR DESCRIPTION
**Pull request by matthiasgeihs**
Closes #368

### CODE REVIEW

File: client/test/backend.go [hunk 1]
```diff
@@ -175,11 +176,12 @@ func (b *MockBackend) Progress(_ context.Context, req channel.ProgressReq) error
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	timeout := time.Now().Add(time.Duration(req.Params.ChallengeDuration) * time.Millisecond)
```
Comment (cryptphil) [inline client/test/backend.go:179]: Shouldn't the ChallengeDuration be given in seconds?

**Review by cryptphil — CHANGES_REQUESTED** [review]
Looks good, just a small remark.

File: client/test/backend.go (hunk 1 above)
Comment (matthiasgeihs) [reply client/test/backend.go:179]: Yes, but this would be too long for the tests. Here we scale down to milliseconds.

Comment (cryptphil) [reply client/test/backend.go:179]: Okay, makes sense.

**Review by cryptphil — APPROVED** [review]

**Pull request merged**